Redesign Timer Prim

daumayr opened the pull request:
This PR adopts the tracing of non-deterministic inputs for the TimerPrims 
to make timed behaviour deterministic in replay executions.
The mechanism of TimerPrims was changed from resolving a Promise to sending a (external) Message that targets a Block.
The Validation replay tests are re-enabled as they now pass.

This solves Issue #262

daumayr committed:
Redesign Timer Prim for Tracing

daumayr committed:
Enable Validation Tests

daumayr committed:
Update Test Data

smarr commented on src/som/primitives/TimerPrims.java:
```diff
@@ -0,0 +1,171 @@
+      timerActor = vm.getMainActor();
+      SInvokable s =
+          (SInvokable) Classes.blockClass.lookupMessage(SELECTOR, AccessModifier.PUBLIC);
+      rct = createOnReceiveCallTarget(SELECTOR,
```
Any reason not to use the `ValueNonePrim` directly here?

Think, I'd just use that. Will perhaps also reduce code complexity here.

smarr replied:
Hm, I see, `AbstractMessageSendNode` is expected by `ReceivedMessage`.

smarr committed:
Simplify changes

- need only one primitive
- initialize timer as part of the actor module init
- avoid use of method references (requires lambda infrastructure)
- use same style for the fields
- initialize structures only in replay mode
- change naming to be more explicit about semantics

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr reviewed (approved):
I simplified the change a little, and change the naming of variables to make the code more explicit.
